core/commands/pin/remotepin.go

```diff
@@ -0,0 +1,612 @@
+package pin
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	neturl "net/url"
+
+	cid "github.com/ipfs/go-cid"
+	cmds "github.com/ipfs/go-ipfs-cmds"
+	config "github.com/ipfs/go-ipfs-config"
+	"github.com/ipfs/go-ipfs/core/commands/cmdenv"
+	fsrepo "github.com/ipfs/go-ipfs/repo/fsrepo"
+	logging "github.com/ipfs/go-log"
+	pinclient "github.com/ipfs/go-pinning-service-http-client"
+	path "github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/libp2p/go-libp2p-core/host"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/multiformats/go-multiaddr"
+)
+
+var log = logging.Logger("core/commands/cmdenv")
+
+var remotePinCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Pin (and unpin) objects to remote pinning service.",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"add":     addRemotePinCmd,
+		"ls":      listRemotePinCmd,
+		"rm":      rmRemotePinCmd,
+		"service": remotePinServiceCmd,
+	},
+}
+
+var remotePinServiceCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Configure remote pinning services.",
+	},
+
+	Subcommands: map[string]*cmds.Command{
+		"add": addRemotePinServiceCmd,
+		"ls":  lsRemotePinServiceCmd,
+		"rm":  rmRemotePinServiceCmd,
+	},
+}
+
+const pinNameOptionName = "name"
+const pinCIDsOptionName = "cid"
+const pinStatusOptionName = "status"
+const pinServiceNameOptionName = "service"
+const pinServiceURLOptionName = "url"
+const pinServiceKeyOptionName = "key"
+const pinBackgroundOptionName = "background"
+const pinForceOptionName = "force"
+
+type RemotePinOutput struct {
+	RequestID string
+	Status    string
+	Cid       string
+	Name      string
+	Delegates []string // multiaddr
+}
+
+func toRemotePinOutput(ps pinclient.PinStatusGetter) RemotePinOutput {
+	return RemotePinOutput{
+		RequestID: ps.GetRequestId(),
+		Name:      ps.GetPin().GetName(),
+		Delegates: multiaddrsToStrings(ps.GetDelegates()),
+		Status:    ps.GetStatus().String(),
+		Cid:       ps.GetPin().GetCid().String(),
+	}
+}
+
+func printRemotePinDetails(w io.Writer, out *RemotePinOutput) {
+	tw := tabwriter.NewWriter(w, 0, 0, 1, ' ', 0)
+	defer tw.Flush()
+	fw := func(k string, v string) {
+		fmt.Fprintf(tw, "%s:\t%s\n", k, v)
+	}
+	fw("RequestID", out.RequestID)
+	fw("CID", out.Cid)
+	fw("Name", out.Name)
+	fw("Status", out.Status)
+}
+
+// remote pin commands
+
+var addRemotePinCmd = &cmds.Command{
```

Comment:
Should not there be an optional `meta` option [as per spec](https://ipfs.github.io/pinning-services-api-spec/#tag/pins/paths/~1pins/post)

> meta - object (PinMeta)  Optional metadata for pin object

Reply:
IIRC we decided to exclude meta from MVP. 
For now name should be enough, `meta` will be added later or in the next release.